splinepy/helpme/create.py

```diff
@@ -298,6 +298,301 @@ def revolved(
+    set_on_trajectory : bool
+      If True, the cross-section will be placed at the evaluation
+      points of the trajectory's knots. If False, the cross-section
+      will be placed at the control points of the trajectory.
+      Default is False.
+    rotation_adaption : float
+      Angle in radians by which the cross-section is rotated around
+      the trajectory tangent vector. This is an additional rotation
+      if the user wants to adapt the cross-section rotation.
+      Example with rectangular cross-section:
+      .         x
+      .       x   x                                      x x x x x
+      .     x       x                                    x       x
+      .   x           x    --> rotation around pi/4 -->  x       x
+      .     x       x                                    x       x
+      .       x   x                                      x x x x x
+      .         x
+
+    Returns
+    -------
+    swept_spline : Spline
+      Spline resulting from the sweep
+    """
+
+    from splinepy import NURBS as _NURBS
+    from splinepy import BSpline as _BSpline
+
+    ### INPUT CHECKS ###
+
+    if not isinstance(cross_section, (_BSpline, _NURBS)):
+        raise TypeError(
+            "cross_section must be an instance of BSpline or NURBS"
+        )
+    if not isinstance(trajectory, (_BSpline, _NURBS)):
+        raise TypeError("trajectory must be an instance of BSpline or NURBS")
+    if not trajectory.para_dim == 1:
+        raise TypeError("trajectory must be of parametric dimension 1")
+    if not isinstance(set_on_trajectory, bool):
+        raise TypeError("set_on_trajectory must be a boolean")
+    if not isinstance(rotation_adaption, (float, int, type(None))):
+        raise TypeError("rotation_adaption must be a float or int")
+
+    if cross_section_normal is not None and not len(cross_section_normal) == 3:
+        raise ValueError("cross_section_normal must be a 3D vector")
+    # setting default value for cross_section_normal
+    if cross_section_normal is None:
+        cross_section_normal = _np.array([0, 0, 1])
+        # add debug message
+        _log.debug("No cross_section_normal given. Defaulting to [0, 0, 1].")
+
+    ### STARTING CALCULATIONS ###
+
+    # make copies so we can work on it inplace
+    trajectory = trajectory.create.embedded(3)
+    cross_section = cross_section.create.embedded(3)
+
+    # initialize parameter values
+    par_value = trajectory.greville_abscissae()
+    par_value = par_value.reshape(-1, 1)
+
+    ### TRANSFORMATION MATRICES ###
+
+    # tangent vector 'e1' of trajectory at parameter value 0
+    e1 = trajectory.derivative([par_value[0]], [1])
+    e1 = (e1 / _np.linalg.norm(e1)).ravel()
+
+    # evaluating a vector normal to e1
+    vec = [-e1[1], e1[0], -e1[2]]
+    B = []
+    # avoid dividing by zero
+    if _np.linalg.norm(_np.cross(e1, vec)) > _settings.TOLERANCE:
+        B.append(_np.cross(e1, vec) / _np.linalg.norm(_np.cross(e1, vec)))
```

Comment:
you can only compute cross once here instead

Reply:
Done via a temp variable